lib/client/base.js

```diff
@@ -745,9 +745,16 @@
   HappnClient.prototype.reconnect = function (options) {
     var _this = this;
 
+    if (!this.reconnectAttempt) this.reconnectAttempt = 0;
+    var reconnectAttempt = ++this.reconnectAttempt;
+
     _this.authenticate(function (e) {
+      if (_this.reconnectAttempt != reconnectAttempt) return; // this is not the last attempt, ignore it
 
-      if (e) return _this.handle_error(e, 3);
+      if (e) {
+        if (e.message && e.message.indexOf('api request timed out action') == 0) return _this.reconnect();
```

Comment:
Can't do setImmediate, this is used in the browser as well. The function is async, so don't see this as a possibility.
Adding a test for 2.